packages/dumb/src/components/trends/common/focused-range-labels/focused-range-labels.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { type FunctionComponent } from 'react'
+
+import { type BgUnit } from 'medical-domain'
+
+import styles from './focused-range-labels.css'
+import { formatClocktimeFromMsPer24 } from '../../../../utils/datetime/datetime.util'
+import { type RangeSegmentSlice } from '../../../../models/enums/range-segment.enum'
+import { formatBgValue } from '../../../../utils/format/format.util'
+import Tooltip from '../../../tooltips/common/tooltip/tooltip'
+
+interface FocusedRangeLabelsProps {
+  bgUnit: BgUnit
+  focusedRangeSegments: RangeSegmentSlice
+  data: {
+    [key in RangeSegmentSlice]: number
+  } & {
+    median: number
+    msFrom: number
+    msTo: number
+    outOfRangeThresholds: {
+      low: number
+      high: number
+    }
+  }
+  position: {
+    left: number
+    tooltipLeft: boolean
+    yPositions: {
+      [key in RangeSegmentSlice]: number
+    } & {
+      topMargin: number
+    }
+  }
+}
```

Comment:
I suggest to factorise this as a type, it will be more readable a little bit more readable and no repeated
```suggestion
type RangeSegmentSliceNumber = {
  [key in RangeSegmentSlice]: number
}
interface FocusedRangeLabelsProps {
  bgUnit: BgUnit
  focusedRangeSegments: RangeSegmentSlice
  data: RangeSegmentSliceNumber & {
    median: number
    msFrom: number
    msTo: number
    outOfRangeThresholds: {
      low: number
      high: number
    }
  }
  position: {
    left: number
    tooltipLeft: boolean
    yPositions: RangeSegmentSliceNumber & {
      topMargin: number
    }
  }
}
```